js/query.js

```diff
@@ -1,70 +1,452 @@
-                                response.status);
-                            return;
-                        }
-
-                        // Examine the text in the response
-                        response.json().then(function(data) {
-                            console.log(data);
-                            var summary = data.summary
-                            document.getElementById('summary').textContent = 'Gene Summary: ' + summary;
-                        });
+    function hideData(divObj) {
+        // Hide all data in child nodes of givin div element.
+        var labels = divObj.querySelectorAll("label");
+        var links = divObj.querySelectorAll('a');
+        var i;
+        var textDivs = divObj.getElementsByClassName("text");
+
+        for (i = 0; i < labels.length; i++) {
+            var childData = labels[i];
+
+            if (!(childData.classList.contains('hidden'))) {
+                childData.classList.add('hidden');
+            }
+        }
+
+        for (i = 0; i < textDivs.length; i++) {
+            var childData = textDivs[i];
+
+            if (!(childData.classList.contains('hidden'))) {
+                childData.classList.add('hidden');
+                childData.textContent = '';
+            }
+        }
+
+        // Delete any links if necessary.
+        for (i = 0; i < links.length; i++) {
+            var childData = links[i];
+            childData.remove();
+        }
+    }
+
+    function annotateGene(gene) {
+        fetch('https://mygene.info/v3/gene/' + gene)
+            .then(
+                function(response) {
+                    if (response.status !== 200) {
+                        console.log('Looks like there was a problem. Status Code: ' +
+                            response.status);
+                        return;
                     }
-                )
-                .catch(function(err) {
-                    console.error('Fetch Gene Annotation Error', err);
+
+                    // Examine the text in the response
+                    response.json().then(function(data) {
+                        console.log(data);
+                        parseGeneData(data).then(function(data) {
+                            displayData(data)
+                        });
+                    });
+                }
+            )
+            .catch(function(err) {
+                console.error('Fetch Gene Annotation Error', err);
+            });
+    }
+
+    function parseGeneData(data) {
+        return new Promise(function(resolve, reject) {
+            var aliases;
+            var hgnc;
+            var coords;
+            var hg19coords;
+            var mm9coords;
+            var names;
+            var uniprotSum;
+            var wiki;
+            var omim;
+            var ensembl;
+            var vega;
+            var pfam;
+            var uniprot;
+            var pharmgkb;
+            var prosite;
+
+            if (data.hasOwnProperty('HGNC')) {
```

Comment:
Initialize all these variables to `null` and remove the `else` clauses below.

Looking at this and the code that follows, I suggest replacing these with a single object containing a key for each variable. This will avoid the redundancy seen in the `resolve` calls down the line.